### PR TITLE
fix : sla callback notification appropriate schedule time

### PR DESCRIPTION
**Pull request by Mryashbhardwaj**

### CODE REVIEW

File: ext/scheduler/airflow/__lib.py [hunk 1]
```diff
@@ -27,15 +27,6 @@
 
 SCHEDULER_ERR_MSG = "scheduler_error"
 
-JOB_START_EVENT_NAME = "job_start_event"
```
Comment (sravankorumilli) [inline ext/scheduler/airflow/__lib.py:30]: i believe these are all unused variables

File: core/scheduler/service/job_run_service.go [hunk 2]
```diff
@@ -436,7 +437,11 @@ func (s *JobRunService) updateOperatorRun(ctx context.Context, event *scheduler.
 
 func (s *JobRunService) trackEvent(event *scheduler.Event) {
 	if event.Type.IsOfType(scheduler.EventCategorySLAMiss) {
-		s.l.Info("received job sla_miss event, jobName: %v , slaPayload: %#v", event.JobName, event.SLAObjectList)
+		jsonSLAObjectList, err := json.Marshal(event.SLAObjectList)
+		if err != nil {
+			jsonSLAObjectList = []byte("unable to json Marshal SLAObjectList")
```
Comment (sravankorumilli) [inline core/scheduler/service/job_run_service.go:442]: log & return immediately and please make all else if as, just if's and return

**Review by sravankorumilli — CHANGES_REQUESTED** [review]

File: ext/scheduler/airflow/__lib.py (hunk 1 above)
Comment (Mryashbhardwaj) [reply ext/scheduler/airflow/__lib.py:30]: yes

File: core/scheduler/service/job_run_service.go (hunk 2 above)
Comment (Mryashbhardwaj) [reply core/scheduler/service/job_run_service.go:442]: actually along with logging we are also tracking these events on the Prometheus , hence we can not return early

Comment (Mryashbhardwaj) [reply core/scheduler/service/job_run_service.go:443]: converted to switch case

**Review by sravankorumilli — APPROVED** [review]

**Pull request merged**